Frameworks/CoreText/CTFrame.mm

```diff
@@ -55,7 +55,7 @@ CFRange CTFrameGetVisibleStringRange(CTFrameRef frame) {
     _CTFrame* framePtr = static_cast<_CTFrame*>(frame);
     CFIndex count = 0;
     if (framePtr) {
-        for (CFIndex i = 0; i < framePtr->_lineOrigins.size() && framePtr->_lineOrigins[i].y > 0; ++i) {
+        for (CFIndex i = 0; i < [framePtr->_lines count] && static_cast<_CTLine*>(framePtr->_lines[i])->_lineOrigin.y >= 0; ++i) {
```

Comment:
>[framePtr->_lines count] && static_cast<_CTLine*>(framePtr->_lines[i])->_lineOrigin.y >= 0 [](start = 32, length = 90)

why the y >= 0? Also, it might make sense to split this inside the for loop so you can get rid of the objectAtIndex call below.